modules/installation-cloudformation-control-plane.adoc

```diff
@@ -0,0 +1,386 @@
+// Module included in the following assemblies:
+//
+// * installing/installing_aws_upi/installing-aws-upi.adoc
+
+[id="installation-cloudformation-control-plane-{context}"]
+= CloudFormation template for control plane machines
+
+You can use the following CloudFormation template to deploy the control plane
+machines that you need for your {product-title} cluster.
+
+[source,yaml]
+----
+AWSTemplateFormatVersion: 2010-09-09
+Description: Template for Openshift Cluster UPI Node Launch (EC2 master instances)
+
+Parameters:
+  InfrastructureName:
+    AllowedPattern: ^([a-zA-Z][a-zA-Z0-9\-]{0,31})$
+    MaxLength: 32
+    MinLength: 1
+    ConstraintDescription: Infrastructure name must be alphanumeric, start with a letter and a maximum of 32 characters
+    Description: A short, unique cluster ID used to tag cloud resources and identify items owned/used by the cluster.
+    Type: String
+  RhcosAmi:
+    Description: Current RHEL CoreOS AMI to use for boostrap
+    Type: AWS::EC2::Image::Id
+  AutoRegisterDNS:
+    Default: "yes"
+    AllowedValues:
+    - "yes"
+    - "no"
+    Description: Do you want to invoke DNS etcd registration (requires Hosted Zone info provided)?
+    Type: String
+  PrivateHostedZoneId:
+    Description: The Route53 private zone ID to register the etcd targets with (e.g Z21IXYZABCZ2A4)
+    Type: String
+  PrivateHostedZoneName:
+    Description: The Route53 zone to register the targets with (No trailing dot - e.g. cluster.mycorp.com)
+    Type: String
+  Master0Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  Master1Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  Master2Subnet:
+    Description: The subnets (recommend private) to launch the master nodes into
+    Type: AWS::EC2::Subnet::Id
+  MasterSecurityGroupId:
+    Description: The master security group ID to associate with master nodes.
+    Type: AWS::EC2::SecurityGroup::Id
+  IgnitionLocation:
+    Default: https://api.$CLUSTER_NAME.$DOMAIN:22623/config/master
+    Description: Location to fetch bootstrap ignition from. (Recommend to use the autocreated ignition config location.)
+    Type: String
+  CertificateAuthorities:
+    Default: data:text/plain;charset=utf-8;base64,ABC...xYz==
+    Description: Base64 encoded certificate authority string to use.
+    Type: String
+  MasterInstanceProfileName:
+    Description: IAM profile to associate with master nodes.
+    Type: String
+  MasterInstanceType:
+    Default: m4.xlarge
+    Type: String
+    AllowedValues:
```

Comment:
Latest generation of EC2 instances aren't allowed (m5, c5, r5).  It might be interesting to allow AMD based instances as well.

Reply:
Correct. There are still outstanding issues with nitro-based instance classes.

https://bugzilla.redhat.com/show_bug.cgi?id=1658237

https://github.com/coreos/bugs/issues/2371

cc @gnufied

Reply:
The installer currently falls...forward to m5 for regions that lack m4.  So if we don't support m5, we'll need to adjust that.

Reply:
https://github.com/openshift/installer/pull/1163#discussion_r257803519

Reply:
Actually, The most serious of NVMe issues are fixed in - https://bugzilla.redhat.com/show_bug.cgi?id=1632440 . Our SRE team has confirmed that after deploying fixed kernel they did not receive any "stuck in attach/detach" issue.  

What @cuppett linked above though is  a separate customer running their own Openshift version on AWS.  That bug however is more weird. The steps to reproduce - https://bugzilla.redhat.com/show_bug.cgi?id=1658237#c46 . So basically to reproduce this bug, you have to reboot the node while detach is in progress.  Our Kernel team says - this is a problem with how EBS sends ACPI events to kernel. In practice I think this is less of a problem. We also have some fixes in Openshift to ensure if a node does get stuck with volumes in "attaching/detaching" state, it is removed from scheduling and is tainted.  It must be then drained, rebooted and taint manually removed.

I think we should make sure that - we are running fixed Kernel with proper tuning in RHCOS and start supporting M5/C5 instance types.

Reply:
That is excellent news! We could enable the latest generation instances in the templates.

@abhinavdahiya @crawford Do we think m5 should be our default? The latest generation would be my recommendation in general and continues the trend of improving cost/performance. A downside is they support a total of 28 EBS and ENA attachments where the previous generations could support up to 40 EBS attachments [1].

[1]: https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/volume_limits.html

Reply:
@gnufied It doesn't appear that https://bugzilla.redhat.com/show_bug.cgi?id=1632440 is publicly viewable; could you share some details or open up access to that bug?